apps/log-switch/app.js

```diff
@@ -1,55 +1,79 @@
 'use strict'
-var DISABLE_LEVEL = require('logger').UPLOAD_DISABLE_LEVEL
-var MIN_LEVEL = require('logger').UPLOAD_MIN_LEVEL
-var MAX_LEVEL = require('logger').UPLOAD_MAX_LEVEL
+var levels = require('logger').levels
 var setGlobalUploadLevel = require('logger').setGlobalUploadLevel
 var logger = require('logger')('log-switch')
 var cloudgw = require('@yoda/cloudgw')
 var property = require('@yoda/property')
 var persistKey = 'log.cloud.level'
+var expireKey = 'log.cloud.expire'
+var switchDefaultTimeout = 60 * 1000
+var defaultLevel = levels.info
 
 module.exports = function (activity) {
   activity.on('ready', () => {
-    var level = property.get(persistKey, 'persist')
-    logger.info(`cloud log init level: ${level}`)
+    var level = parseInt(property.get(persistKey, 'persist'))
+    var expire = parseInt(property.get(expireKey, 'persist'))
+    var timeout
+    logger.info(`cloud log init level: ${level}, expire at ${expire}`)
     if (!level) {
-      level = MIN_LEVEL
+      level = defaultLevel
     }
-    onCloudLogLevelSwitch(activity, level)
+    if (expire) {
+      timeout = expire - Date.now()
+      // the previous level was expired, revert to default level
+      if (timeout < 1) {
+        timeout = undefined
+        level = defaultLevel
+      }
+    }
+    onCloudLogLevelSwitch(activity, level, timeout)
   })
   activity.on('request', (nlp, action) => {
     if (nlp.intent === 'RokidAppChannelForward') {
-      var level = nlp.forwardContent.intent
-      onCloudLogLevelSwitch(activity, level)
+      var level = parseInt(nlp.forwardContent.intent)
+      var timeout = parseInt(nlp.forwardContent.slots.timeout)
```

Comment:
`nlp.forwardContent.slots.timeout`. this may cause an error. Maybe use _.get(obj, name) would be better.